### PR TITLE
Setup dev env

### PR DESCRIPTION
**Pull request by VanGrx**
- Adding github action to build, check format and clippy warnings, and test.
- Fix clippy warnings
- Separate GUI part to a separate file

### CODE REVIEW

**Review by matheusfigueredo — DISMISSED** [review]
LGTM! 🚀

**Comment by VanGrx** [comment]
Forced push as it asked to verify commits now???

File: src/main_app.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
+use crossbeam_channel::{select, unbounded, Receiver};
```
Comment (matheusfigueredo) [inline src/main_app.rs:1]: What do you think about moving this file to a proper folder (e.g. `gui`)?

**Review by matheusfigueredo — APPROVED** [review]
LGTM!

Just a small comment, let me know what do you think

File: src/main_app.rs (hunk 1 above)
Comment (matheusfigueredo) [reply src/main_app.rs:1]: Or maybe create a `bin` folder, and the GUI App keeps in another directory.

Anyway, just some thoughts :)

Comment (VanGrx) [reply src/main_app.rs:1]: Agree. But let's first see how the file structure will go and separate in folders if we see clear chance for that.

**Pull request merged**